public/app/core/utils/richHistory.ts

```diff
@@ -60,8 +60,14 @@ export function addToRichHistory(
     ];
 
     /* Combine all queries of a datasource type into one rich history */
-    store.setObject(RICH_HISTORY_KEY, newHistory);
-    return newHistory;
+    const isSaved = store.setObject(RICH_HISTORY_KEY, newHistory);
+
+    /* If newHistory is succesfully saved, return it. Otherwise return not updated richHistory.  */
+    if (isSaved) {
+      return newHistory;
+    } else {
+      return richHistory;
+    }
```

Comment:
Error handling - if object is not saved in store, it returns false. Therefore we need to return newHistory if saving was successful and previous history if it wasn't successful.